connectors/sources/sharepoint_online.py

```diff
@@ -0,0 +1,589 @@
+
+                        if "@microsoft.graph.downloadUrl" in drive_item:
+                            if "size" in drive_item and drive_item["size"] < 10485760:
+                                download_func = partial(self.get_content, drive_item)
+                            else:
+                                print(
+                                    f"Not downloading file {drive_item['name']} of size {drive_item['size']}"
+                                )
+
+                        yield drive_item, download_func
+
+                async for site_list in self._client.site_lists(site["id"]):
+                    site_list["_id"] = site_list["id"]
+                    site_list["object_type"] = "site_list"
+
+                    yield site_list, None
+
+                    async for list_item in self._client.site_list_items(
+                        site["id"], site_list["id"]
+                    ):
+                        list_item["_id"] = list_item["id"]
+                        list_item["object_type"] = "list_item"
+                        content_type = list_item["contentType"]["name"]
+
+                        if (
+                            content_type == "Web Template Extensions"
+                        ):  # TODO: make it more flexible. For now I ignore them cause they 404 all the time
+                            continue
+
+                        if "Attachments" in list_item["fields"]:
+                            async for list_item_attachment in self._client.site_list_item_attachments(
+                                site["webUrl"], site_list["name"], list_item["id"]
+                            ):
+                                list_item_attachment["_id"] = list_item_attachment[
+                                    "odata.id"
+                                ]
+                                list_item_attachment[
+                                    "object_type"
+                                ] = "list_item_attachment"
+                                list_item_attachment["_timestamp"] = list_item[
+                                    "lastModifiedDateTime"
+                                ]
+                                attachment_download_func = partial(
+                                    self.get_attachment, list_item_attachment
+                                )
+                                yield list_item_attachment, attachment_download_func
+
+                        download_func = None
+
+                        yield list_item, download_func
+
+                async for site_page in self._client.site_pages(site["webUrl"]):
+                    site_page["_id"] = site_page["GUID"]
+                    site_page["object_type"] = "site_page"
+                    if "LayoutWebpartsContent" in site_page and site_page["LayoutWebpartsContent"]:
+                        site_page["LayoutWebpartsContent"] = BeautifulSoup(site_page["LayoutWebpartsContent"], features="html.parser").get_text()
```

Comment:
That is going to be cpu-bound and blocking the GIL. Can you put the parsing in a separate function in utils. I'll explore alternatives to see of we can avoid the toll

Reply:
Added a note there -- if this data is an xml-compatible tree you can switch to `lxml` https://github.com/elastic/connectors-python/issues/973

Reply:
Sure, this code is there in WIP state and not indicative of final solution :D

I'll move it to a function in utils and explore the suggestion in #973.